paddle/fluid/framework/tensor.h

```diff
@@ -168,6 +168,8 @@ class Tensor {
 
   void ResetHolder(std::shared_ptr<memory::Allocation> holder);
 
+  void ResetType(const proto::VarType::Type type) { type_ = type; }
```

Comment:
Try to add a function like `void ResetHolderWithType(std::shared_ptr<memory::Allocation> holder, proto::VarType::Type type)` instead of reseting data type alone. Reseting data type alone is meaningless.

In this PR, you would call `ResetHolder` and `ResetType` in the meantime, but it is not guaranteed in the future development.